### ISSUE TITLE
Missing Implicit With Warning for Reports

### ISSUE THREAD

**Issue by pri-kise** [microsoft/AL#6272]
**Describe the bug**
Is you use a function of a Record inside a column of a Report. There isn't any warning for the implicit with usage.

**To Reproduce**
Steps and to reproduce the behavior:
Check out the Std. Sales Order Confirmation Report and have a look at `GetCustomerVATRegistrationNumber()`.

``` AL Code to reproduce the issue ```

**Expected behavior**
We should get a warning/info that implicit with will be an error in the future.
**Screenshots**

**5. Versions:**

 - AL Language: 6.1.359074

 - Business Central:

**Comment by dtkb** [comment]
This should presumably also apply inside procedures.

**Comment by dtkb** [comment]
Are implicit with warnings that _do_ exist only meant to appear when targeting BC17/runtime6, or what? I don't get them anymore for obvious cases where I expect I should in BC16/runtime5 (i.e. where it costs me nothing to add `Rec.` now and avoid work later):

![image](https://user-images.githubusercontent.com/63455896/98387997-0e07e800-204a-11eb-9d10-eb5988a2d561.png)

**Comment by atoader** [comment]
We changed it so that they only appear when targeting runtime "6.0"

**Comment by fvet** [comment]
And apparently only if target is set to Cloud...

**Comment by pri-kise** [comment]
I have changed my project to runtime 6.0 and target cloud, but i still don't get warnings for reports.

**Comment by dtkb** [comment]
Did you close/reopen VS Code? It doesn't seem to honour changes to `Runtime` otherwise...

**Comment by pri-kise** [comment]
Yes i did repoen VS Code to get pragmas to work and they work now, but there is no warning for colums in reports.

**Comment by pri-kise** [comment]
Any progress on this issue?

**Comment by SBalslev** [comment]
Hi! Thank you for this detailed report about missing Implicit With warnings for Reports — the community discussion here was very constructive.

We can confirm this area was addressed internally, including a full Implicit With code action implementation across document, project, and workspace scope (PR 183471), shipped around **Business Central runtime 25.0 (2024 Wave 2)**.

We're closing this as resolved. If you still see missing Implicit With warnings in reports on a current version, please open a new issue.

Thank you for helping us improve the AL developer experience!
